runtime/doc/job_control.txt

```diff
@@ -102,36 +102,29 @@ function. Here's a more object-oriented version of the above:
 >
     let Shell = {}
 
-    function Shell.on_stdout(job_id, data) dict
-      call append(line('$'), self.get_name().' stdout: '.join(a:data))
+    function Shell.on_stdout(_job_id, data, event)
+      call append(line('$'),
```

Comment:
I kind of liked that the user is free to move around while this is happening, but you're right that we should jump to the end at least once.

How about..

```vim
function Shell.new(name, cmd)
  let object = extend(copy(g:Shell), {'name': a:name})
  let object.cmd = ['sh', '-c', a:cmd]
  let object.id = jobstart(object.cmd, object)
  $
  return object
endfunction
```